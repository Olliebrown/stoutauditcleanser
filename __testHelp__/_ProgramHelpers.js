import fs from 'fs'
import path from 'path'

// Project specific imports
import { QUERIES } from '../src/domTraversal/queriesAndRegex.js'
import Program from '../src/Objects/Program.js'

export function InitProgramTest (htmlFileName) {
  // Read in the HTML page
  const baseHTML = fs.readFileSync(
    path.join('__tests__', 'data', 'ProgramTables', htmlFileName),
    'utf8'
  )

  // Load page from imported HTML
  document.documentElement.innerHTML = baseHTML.toString()

  // Extract program headers and make sure there's at least one
  const headerNodes = document.querySelectorAll(QUERIES.programHeader)
  if (headerNodes?.length < 1) {
    throw new Error('Unable to find at least one program')
  }

  // Create program object
  return new Program(headerNodes[0])
}
