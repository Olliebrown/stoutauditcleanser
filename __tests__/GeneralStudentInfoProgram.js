// Necessary when using ECMA modules
import { jest } from '@jest/globals'

// Help load the program HTML page
import { InitProgramTest } from '../__testHelp__/_ProgramHelpers.js'

// Disable jest mocking for fs module
jest.dontMock('fs')

describe('General Student Info Node', () => {
  // Inject clean version of HTML page and rebuild program before tests
  let program = null
  beforeAll(() => {
    // Load page from imported HTML
    program = InitProgramTest('GeneralInfo.html')
  })

  // Clear out the HTML and program
  afterAll(() => {
    jest.resetModules()
    program = null
  })

  // Basic page structure tests
  test('should have a specific page title and specific program name', () => {
    // Check title
    const title = document.querySelector('title')
    expect(title).toBeTruthy()
    expect(title.textContent).toBe('Student General Info')

    // Check program name
    expect(program).toBeTruthy()
    expect(program.getName()).toBe('General Information For Students')
  })
})
