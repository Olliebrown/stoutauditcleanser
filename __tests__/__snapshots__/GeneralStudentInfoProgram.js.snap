// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`General Student Info Node should have expected snapshot 1`] = `
{
  "internalId": "RG-10369",
  "key": "general_information_for_students",
  "name": "General Information For Students",
  "satisfied": "INCOMPLETE",
  "subNodes": [
    {
      "description": "Incomplete/In Progress/Courses in Registration StatusIncomplete Coursework, that has earned a Grade of an "I", must be completed within one year from the date the "I" is received or an "F" Grade will be assigned. A grade of "IP" is given for approved courses in which more than one term is required for completion because of the nature of the course requirement.  If the course requirements are not completed within two years from the date the "IP" is received, the student must re-register for the course to obtain credit. Please take the following courses into consideration when planning your future class registration.Visit the Registration & Records website  for further information.",
      "index": 7,
      "internalId": "RQ-0169",
      "isGenEd": false,
      "key": "incomplete_in_progress_coursework_notice",
      "name": "Incomplete/in-progress Coursework Notice",
      "nextIndex": 15,
      "satisfied": "INCOMPLETE",
      "subNodes": [],
      "units": {},
    },
  ],
  "units": {},
}
`;
